lib/spells/effects/Catapult.cpp

```diff
@@ -106,20 +112,130 @@ void Catapult::apply(ServerCallback * server, const Mechanics * m, const EffectT
 		if (attackInfo == ca.attackedParts.end()) // new part
 		{
 			CatapultAttack::AttackInfo newInfo{};
-			newInfo.damageDealt = 1;
+			newInfo.damageDealt = getRandomDamage(server);
 			newInfo.attackedPart = target;
 			newInfo.destinationTile = m->battle()->wallPartToBattleHex(target);
 			ca.attackedParts.push_back(newInfo);
 			attackInfo = ca.attackedParts.end() - 1;
 		}
 		else // already damaged before, update damage
 		{
-			attackInfo->damageDealt += 1;
+			attackInfo->damageDealt += getRandomDamage(server);
 		}
 	}
-
 	server->apply(&ca);
 
+	removeTowerShooters(server, m);
+}
+
+void Catapult::applySeparate(ServerCallback * server, const Mechanics * m, const EffectTarget & target) const
```

Comment:
Considering that Earthquake and Catapult/Cyclops have basically separate apply logic maybe it is better to have separate effect classes? 

Potentially have one derived from another (or from common abstract class)

Reply:
@IvanSavenko We can do 3 classes, but then it will be 3 effect names. And logic is not separate, there is many common functions, as you can see.

Reply:
@IvanSavenko is there a reason why in earthquake std::set was used, or it can be safely replaced by std::vector?

Reply:
1) 3 classes, 2 effect names. No need to provide abstract class to mods. 
All common functions can be delegated to base abstact class.

std::set - probably no particular reason, other than being the most convenient.

Reply:
@IvanSavenko so, it can be merged too. Only pack then will be different. Will show rework later.

Reply:
most probable std::set intended use was to avoid duplicates for whatever reason

Reply:
@IvanSavenko split more common code from two apply functions. Maybe it will enough to have it a common effect?

Reply:
Personally I would still split it into classes, but not going to insist. So if you wish to keep it this way - OK.